tlslite/utils/codec.py

```diff
@@ -3,31 +3,195 @@
 
 """Classes for reading/writing binary data (such as TLS records)."""
 
-from .compat import *
+import sys
+import struct
+from struct import pack
+
 
 class Writer(object):
+    """Serialisation helper for complex byte-based structures."""
+
     def __init__(self):
+        """Initialise the serializer with no data."""
         self.bytes = bytearray(0)
 
-    def add(self, x, length):
-        self.bytes += bytearray(length)
-        newIndex = len(self.bytes) - 1
-        for count in range(length):
-            self.bytes[newIndex] = x & 0xFF
-            x >>= 8
-            newIndex -= 1
-        if x != 0:
-            raise ValueError("Can't represent value in specified length")
+    def addOne(self, val):
+        """Add a single-byte wide element to buffer, see add()."""
+        self.bytes.append(val)
+
+    if sys.version_info < (2, 7):
```

Comment:
I'm not a fan of this change. There's nothing wrong with it outside of a ton of complexity. Is the speedup really worth how much more complex this is? We could write our own pack method to fix this instead of doing it this way perhaps.